src/sentry/middleware/integrations/tasks.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from typing import Any, Dict, List, MutableMapping, cast
+
+import requests
+import sentry_sdk
+from requests import Response
+
+from sentry.models.outbox import ControlOutbox
+from sentry.silo.base import SiloMode
+from sentry.silo.client import RegionSiloClient
+from sentry.tasks.base import instrumented_task
+from sentry.types.region import get_region_by_name
+from sentry.utils import json
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.middleware.integrations.tasks.convert_to_async_slack_response",
+    queue="integrations.control",
+    silo_mode=SiloMode.CONTROL,
+)
+def convert_to_async_slack_response(
```

Comment:
While these two tasks are very similar, they have a few differences. Instead of DRYing up the shared parts, i kept them repeated since theres no reason these two integrations don't diverge in patterns down the line.